### PR TITLE
[PARTIAL-4] refactor links and show retry button

### PR DESCRIPTION
**Pull request by pregnantboy**
It will render the resend button based on the error details

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#623** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/623?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#615** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/615?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#614** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/614?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#613** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/613?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#612** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/612?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#611** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/611?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by m0nggh — APPROVED** [review]
lgtm, just checking that the `what's new` red dot is intended to always be there right?

File: packages/frontend/src/components/ExecutionStep/index.tsx [hunk 1]
```diff
@@ -30,20 +31,29 @@ type ExecutionStepProps = {
...
-const errorIcon = (
+const failureIcon = (
   <Icon
     boxSize={6}
     as={BiSolidErrorCircle}
     color="interaction.critical.default"
   />
 )
+
+const partialIcon = (
```
Comment (m0nggh) [inline packages/frontend/src/components/ExecutionStep/index.tsx:49]:
```suggestion
const partialSuccessIcon = (
```

**Comment by pregnantboy** [comment]
### Merge activity

* **Jun 18, 2:55 AM EDT**: @pregnantboy started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/614).
* **Jun 18, 3:02 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/614) rebased this pull request as part of a merge.
* **Jun 18, 3:03 AM EDT**: @pregnantboy merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/614).

**Pull request merged**